notebooks/week2/05.log_and_register_fe_model.py

```
@@ -0,0 +1,214 @@
+    # Log model parameters, metrics, and model
+    mlflow.log_param("model_type", "RandomForestClassifier with preprocessing")
+    mlflow.log_params(parameters)
+    mlflow.log_metric("score", accuracy)
+    signature = infer_signature(model_input=X_train, model_output=y_pred)
+
+    # Log model with feature engineering
+    fe.log_model(
+        model=pipeline,
+        flavor=mlflow.sklearn,
+        artifact_path="randomforest-pipeline-model-fe",
+        training_set=training_set,
```

Ruff

Undefined name `training_set`

(F821)